datasets/superb/superb.py

```diff
@@ -169,24 +206,189 @@ def _split_generators(self, dl_manager):
-                    speaker_id, chapter_id = [int(el) for el in id_.split("-")[:2]]
+        if self.config.name == "asr":
+            transcripts_glob = os.path.join(archive_path, "LibriSpeech", "*/*/*/*.txt")
+            key = 0
+            for transcript_path in sorted(glob.glob(transcripts_glob)):
+                transcript_dir_path = os.path.dirname(transcript_path)
+                with open(transcript_path, "r", encoding="utf-8") as f:
+                    for line in f:
+                        line = line.strip()
+                        id_, transcript = line.split(" ", 1)
+                        audio_file = f"{id_}.flac"
+                        speaker_id, chapter_id = [int(el) for el in id_.split("-")[:2]]
+                        yield key, {
+                            "id": id_,
+                            "speaker_id": speaker_id,
+                            "chapter_id": chapter_id,
+                            "file": os.path.join(transcript_dir_path, audio_file),
+                            "text": transcript,
+                        }
+                        key += 1
+        elif self.config.name == "sd":
+            data = SdData(archive_path)
+            args = SdArgs()
+            chunk_indices = _generate_chunk_indices(data, args, split=split)
+            if split != "test":
+                for key, (rec, st, ed) in enumerate(chunk_indices):
+                    speakers = _get_speakers(rec, data, args)
                     yield key, {
-                        "id": id_,
-                        "speaker_id": speaker_id,
-                        "chapter_id": chapter_id,
-                        "file": os.path.join(transcript_dir_path, audio_file),
-                        "text": transcript,
+                        "record_id": rec,
+                        "file": data.wavs[rec],
+                        "start": st,
+                        "end": ed,
+                        "speakers": speakers,
                     }
-                    key += 1
+            else:
+                key = 0
+                for rec in chunk_indices:
+                    for rec, st, ed in chunk_indices[rec]:
+                        speakers = _get_speakers(rec, data, args)
+                        yield key, {
+                            "record_id": rec,
+                            "file": data.wavs[rec],
+                            "start": st,
+                            "end": ed,
+                            "speakers": speakers,
+                        }
+                        key += 1
+
+
+class SdData:
+    def __init__(self, data_dir):
+        """Load sd data."""
+        self.segments = self._load_segments_rechash(data_dir["segments"])
+        self.utt2spk = self._load_utt2spk(data_dir["utt2spk"])
+        self.wavs = self._load_wav_zip(data_dir["wav.zip"])
+        self.reco2dur = self._load_reco2dur(data_dir["reco2dur"])
+        # self.spk2utt = self._load_spk2utt(data_dir["spk2utt"])
```

Comment:
remove?

Reply:
Yes, I'm removing this as we do not use it to generate the dataset.